### ISSUE TITLE
[Core][Pet] Pet interactions with stealthed units

### ISSUE THREAD

**Issue by Zarant** [Atlantiss/NetherwingBugtracker#3695]
Currently, pets cannot interact with stealthed units at all, even if you have hunter's mark active on the target:

![axzi1xgvy0](https://user-images.githubusercontent.com/26726709/50523031-16d61d00-0ab5-11e9-9378-f261effaf927.gif)

Pets should be able to interact with stealthed units as long as the master have vision on them.
You can see on this clip from the warlock PoV, he finds the rogue and gets sapped immediately after, then his succubus manages to cast seduction, even when the succubus herself was outside of the stealth detection radius:
https://youtu.be/EQ_rrjzYImg?t=435

And to add some extra context, after patch 2.4.3, pets were no longer chasing players after casting stealth, however, I could dig a few testimonies from players about this particular behavior:

> Not sure if this has already been stated but testing it with a tauren friend of mine stealthing as I send my pet after him I found that if your pet is on Agressive/Defensive that he'll sit right at the stealthed target but wont attack the target **until you can see him**. It's sort of odd. 

> Tested today guys.
> 
> Results are:
> 
> Send pet on druid. Druid shadowmelds, pet immediately turns around and goes back to me, he comes out, it turns back around.
> 
> If I hunters mark, then the pet will ignore his shadowmeld and go straight to him.
> 
> So just keep up a hunters mark and theres not much to worry about.
> 
> But still a very gay change. 

http://www.arenajunkies.com/topic/29172-hidden-hunter-nerfbug-fix-in-243/page-2

So to sum up, based on those 2 testimonies from people that tested the mechanic during patch 2.4.3, you can see that pets cannot attack units that you have no vision over, even on agressive mode, the pet was following the target around but couldn't attack it until the master had vision over that unit. Another interesting bit is that when you lose vision over an unit, your pet continues to execute his normal AI routine, despite not being able to attack that unit.

Server revision: 2515

**Comment by killablink** [comment]
Yeah, pets should obviously be able to attack stealthed targets that the owner is able to see. And also confirm that pets should stop following a target if the target stealths (and the owner is too far to see him stealth)

**Comment by Brasso** [comment]
Also reported here #1415 https://www.youtube.com/watch?v=dqQ2Rt54RCI

**Comment by Easyfraggy** [comment]
https://github.com/Atlantiss/NetherwingBugtracker/issues/1415
